controllers/sr_controller/sr_controller.py

```diff
@@ -7,25 +7,57 @@
 # Root directory of the SR webots simulator (equivalent to the root of the git repo)
 ROOT = Path(__file__).resolve().parent.parent.parent
 
-ROBOT_FILE = ROOT.parent / "robot.py"
+MODE_FILE = ROOT / "robot_mode.txt"
 
 EXAMPLE_CONTROLLER_FILE = ROOT / "controllers/example_controller/example_controller.py"
 
+ROBOT_IDS_TO_CORNERS = {
+    "291": 0,
+    "684": 1,
+    "1077": 2,
+    "1470": 3,
+}
 
-if __name__ == "__main__":
-    if not ROBOT_FILE.exists():
+def get_robot_zone() -> int:
+    return ROBOT_IDS_TO_CORNERS[os.environ['WEBOTS_ROBOT_ID']]
+
+def get_robot_file() -> Path:
+    zone_id = get_robot_zone()
+    robot_file_dir = ROOT.parent / str(zone_id)
+    if robot_file_dir.is_dir():
+        return robot_file_dir / "robot.py"
+
+    return ROOT.parent / "robot.py"
+
+def get_robot_mode() -> str:
+    if not MODE_FILE.exists():
+        return "dev"
+    return MODE_FILE.read_text().strip()
+
+
+def main():
+    robot_file = get_robot_file()
+
+    if not robot_file.exists():
         print("Robot controller not found, copying example into place.")
-        copyfile(str(EXAMPLE_CONTROLLER_FILE), str(ROBOT_FILE))
+        copyfile(str(EXAMPLE_CONTROLLER_FILE), str(robot_file))
```

Comment:
:+1: to not copying in comp mode, that's a neat solution.

I don't think it's a good idea to exit cleanly for what is essentially an error though. I suggest that we have the controller error loudly in comp mode if it can't find the `robot.py`. I think we should explicitly signal that we expect no robot in a given zone.

I am very much expecting that we'll also remove robots from the simulation if there isn't a team in that corner, which is what we'd do in a real competition.

Reply:
acf9e5b6e33d6c391b70211c53fd18cdd6070e67 (merges https://github.com/srobo/competition-simulator/pull/106) reworked this substantially.